### ISSUE TITLE
[3.0]: Time format needs to accept a format.

### ISSUE THREAD

**Issue by jdarwood007** [SimpleMachines/SMF#7985]
### Basic Information

# Expected behavior:
-------------------
No parsing error:

# Actual behavior:
-------------------
```
				return '[time]' . Time::create('@' . $matches[1])->setTimezone(new \DateTimeZone(User::getTimezone()))->format(null, false) . '[/time]';
```
This results in the following errors:
```
Argument '1' passed to format() is expected to be of type string, null given
Too many arguments to function format(). 2 provided, but 1 accepted.
```

This is occurring because setTimezone returns DateTimeZone, which the format() method expects a $format

### Version/Git revision

0fd130ded

### Database Engine

All

**Comment by Sesquipedalian** [comment]
1. What file is this in?
2. Can we have a better title?

**Comment by Oldiesmann** [comment]
> 1. What file is this in?

Sources/Msg.php line 949.

**Comment by jdarwood007** [comment]
Added a fix in PR #7988.  I think it will work

The issue is that setTimezone returns a datetime object, not a SMF\Time object, so the format method needs the format specified as the first argument.  I did a fix by calling the Time::getDateFormat() as the argument.  Another option to fix would be for SMF\Time to overload the DateTime::setTimezone just to return a SMF\Time, which then would allow the SMF\Time::format() to work.
